Actions/PullRequestClosedAction.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using Slack.Json.Github;
+using Slack.Json.Slack;
+using Slack.Json.Util;
+
+namespace Slack.Json.Actions
+{
+    public class PullRequestClosedAction : IRequestAction
+    {
+        public string GithubHookEventName => "pull_request";
+        public string SlackJsonType => "pull_request";
+        private readonly ISlackMessaging slack;
+        private readonly ILogger<PullRequestClosedAction> logger;
+
+        public PullRequestClosedAction(ISlackMessaging slack, ILogger<PullRequestClosedAction> logger)
+        {
+            this.slack = slack;
+            this.logger = logger;
+        }
+
+        public void Execute(JObject request, IEnumerable<ISlackAction> actions)
+        {
+            if(request.Get<string>(x => x.action) != "closed")
+                return;
+
+            ActionUtils.ParsePullRequestDefaultFields(request, out var prHtmlUrl, out var prTitle);
```

Comment:
Outs could be avoided here (if you have time)
```
        public static (string prHtmlUrl, string prTitle) ParsePullRequestDefaultFields(JObject request)
        {
            var prHtmlUrl = request["pull_request"]?["html_url"].Value<string>()
                ?? throw new InvalidOperationException($"JSON is missing pull request html_url: {request}");
            var prTitle = request["pull_request"]?["title"].Value<string>()
                ?? throw new InvalidOperationException($"JSON is missing pull request title: {request}");
            return (prHtmlUrl, prTitle);
        }
```

Reply:
Valid point, however i think i'll skip refactoring at this point since that util method is used in many locations and i don't have currently time to invest.